packages/libraries/core/src/client/http-client.ts

```diff
@@ -132,26 +166,26 @@ export async function makeFetchCall(
             getErrorMessage(error);
 
           if (isFinalAttempt) {
-            logger?.error(msg);
+            logger.error({ error }, msg);
             return;
           }
-          logger?.debug?.(msg);
+          logger.debug({ error }, msg);
         };
 
         if (isAggregateError(error)) {
           for (const err of error.errors) {
             if (isFinalAttempt) {
-              logger?.error(err);
+              logger.error({ error: err });
               continue;
             }
-            logger?.debug?.(String(err));
+            logger.debug(String(err));
           }
 
           logErrorMessage();
           throw new Error(`Unexpected HTTP error. (x-request-id=${requestId})`, { cause: error });
         }
 
-        logger?.error(error);
+        logger.error({ error }, String(error));
```

Comment:
```suggestion
        logger.error({ error });
```

Reply:
Would the legacy logger implementation in that case log anything here if there is no message? Or is the message auto-constructed from attributes if it includes an error object?